// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "com.github.nholbrook.foodfinder";
option java_outer_classname = "FoodFinderProto";
option objc_class_prefix = "FF";

package foodfinder;

service FoodFinder {
  rpc RequestSupplyInfo (SupplyRequest) returns (SupplyInfo) {}
}

service Supplier {
  rpc RequestVendorInfo (SupplyRequest) returns (stream VendorInfo) {}
}

service Vendor {
  rpc RequestInventoryInfo (SupplyRequest) returns (InventoryInfo) {}  
}

message SupplyRequest {
  string name = 1;
  uint32 quantity = 2;
}

message SupplyInfo {
  InventoryInfo inventory = 1;
  VendorInfo vendor = 2;
}

message VendorInfo {
  string url = 1;
  string name = 2;
}

message InventoryInfo {
  float price = 1;
  uint32 quantity = 2;
}